### PR TITLE
Input form with debounce

### LINKED ISSUE #1335: [TestSuitePermissions] autocompletion when selecting organization
When user wants to give permissions over test suites to another organization, he might want to have an autocompletion.

### PR DESCRIPTION
**Pull request by sanyavertolet**
This PR closes #1335

### What's done:
 - [x] Moved input form with debounced autocompletion to the new component
 - [x] Reused new component in manageTestSuitePermissionsComponent

### CODE REVIEW

File: save-backend/src/main/kotlin/com/saveourtool/save/backend/service/OrganizationService.kt [hunk 1]
```diff
@@ -101,6 +101,20 @@ class OrganizationService(
     fun getByName(name: String) = findByName(name)
         ?: throw NoSuchElementException("There is no organization with name $name.")
 
+    /**
+     * @param prefix prefix of organization name
+     * @param status status of required organizations
+     * @return list of organizations with required [status] which names start with [prefix]
+     */
+    fun getByPrefixAndStatus(prefix: String, status: OrganizationStatus) = if (prefix.isBlank()) {
```
Comment (Cheshiriks) [inline save-backend/src/main/kotlin/com/saveourtool/save/backend/service/OrganizationService.kt:109]: Let's delete the getNotDeletedOrganizations function now. And we use your function. This is the same function as this one, but with hardcoded stats.

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/basic/ManageUserRoleCard.kt [hunk 2]
```diff
@@ -194,27 +170,20 @@ private fun manageUserRoleCardComponent() = FC<ManageUserRoleCardProps> { props
...
-                    for (user in usersNotFromGroup) {
-                        option {
-                            value = user.name
-                            label = user.source ?: ""
+                    decodeListFromJsonString = { it.decodeFromJsonString() }
+                    getHTMLDataListElementFromOption = { childrenBuilder, userInfo ->
+                        with(childrenBuilder) {
+                            option {
+                                value = userInfo.name
+                                label = userInfo.source
```
Comment (Cheshiriks) [inline save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/basic/ManageUserRoleCard.kt:185]: Is a source needed?

**Review by Cheshiriks — CHANGES_REQUESTED** [review]

File: save-backend/src/main/kotlin/com/saveourtool/save/backend/service/OrganizationService.kt (hunk 1 above)
Comment (sanyavertolet) [reply save-backend/src/main/kotlin/com/saveourtool/save/backend/service/OrganizationService.kt]: Done

**Review by petertrr — APPROVED** [review]

**Review by Cheshiriks — APPROVED** [review]

**Review by akuleshov7 — APPROVED** [review]

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/inputform/InputWithDebounce.kt [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+    /**
+     * Callback to set selected option
+     */
+    var setSelectedOption: (T) -> Unit
+
+    /**
+     * Callback to create [option] tag from [T]
+     */
+    @Suppress("VARIABLE_NAME_INCORRECT_FORMAT", "TYPE_ALIAS")
+    var getHTMLDataListElementFromOption: (ChildrenBuilder, T) -> Unit
```
[the remark this answers is not in the document]
Comment (sanyavertolet) [reply save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/inputform/InputWithDebounce.kt:68]: Yep, but I would like it to be `HTML` to be more like the original type (`HTMLDataListElement`)

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/inputform/InputWithDebounce.kt [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+val inputWithDebounceForString = inputWithDebounce<String>()
+
+/**
+ * [Props] for [inputWithDebounce] component
+ */
+external interface InputWithDebounceProps<T> : Props {
+    /**
+     * Callback to get url for options fetch
+     */
+    var getUrlForOptions: (String) -> String
```
[the remark this answers is not in the document]
Comment (sanyavertolet) [reply save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/inputform/InputWithDebounce.kt]: Added parameter names as far as I believe that might be enough to understand

**Pull request merged**